parquet-variant-compute/src/variant_parser.rs

```diff
@@ -0,0 +1,669 @@
+                buffer.extend_from_slice(&bytes[..3]);
+            }
+            4 => buffer.extend_from_slice(&(value as u32).to_le_bytes()),
+            _ => panic!("Invalid size: {}", size),
+        }
+    }
+
+
+
+    /// Check if value bytes represent a primitive
+    pub fn is_primitive(value_bytes: &[u8]) -> bool {
+        if value_bytes.is_empty() {
+            return false;
+        }
+        Self::get_basic_type(value_bytes[0]) == VariantBasicType::Primitive
+    }
+
+    /// Check if value bytes represent a short string
+    pub fn is_short_string(value_bytes: &[u8]) -> bool {
+        if value_bytes.is_empty() {
+            return false;
+        }
+        Self::get_basic_type(value_bytes[0]) == VariantBasicType::ShortString
+    }
+
+    /// Check if value bytes represent an object
+    pub fn is_object(value_bytes: &[u8]) -> bool {
+        if value_bytes.is_empty() {
+            return false;
+        }
+        Self::get_basic_type(value_bytes[0]) == VariantBasicType::Object
+    }
+
+    /// Check if value bytes represent an array
+    pub fn is_array(value_bytes: &[u8]) -> bool {
+        if value_bytes.is_empty() {
+            return false;
+        }
+        Self::get_basic_type(value_bytes[0]) == VariantBasicType::Array
+    }
+
+    /// Get the data length for a primitive type
+    /// Returns Some(len) for fixed-length types, None for variable-length types
+    pub fn get_primitive_data_length(primitive_type: &PrimitiveType) -> Option<usize> {
+        match primitive_type {
+            PrimitiveType::Null | PrimitiveType::True | PrimitiveType::False => Some(0),
+            PrimitiveType::Int8 => Some(1),
+            PrimitiveType::Int16 => Some(2),
+            PrimitiveType::Int32
+            | PrimitiveType::Float
+            | PrimitiveType::Decimal4
+            | PrimitiveType::Date => Some(4),
+            PrimitiveType::Int64
+            | PrimitiveType::Double
+            | PrimitiveType::Decimal8
+            | PrimitiveType::TimestampNtz
+            | PrimitiveType::TimestampLtz => Some(8),
+            PrimitiveType::Decimal16 => Some(16),
+            PrimitiveType::Binary | PrimitiveType::String => None, // Variable length, need to read from data
+        }
+    }
+
+    /// Extract short string data from value bytes
+    pub fn extract_short_string_data(value_bytes: &[u8]) -> Result<&[u8], ArrowError> {
+        if value_bytes.is_empty() {
+            return Err(ArrowError::InvalidArgumentError(
+                "Empty value bytes".to_string(),
+            ));
+        }
+
+        let header = Self::parse_short_string_header(value_bytes[0])?;
+
+        if value_bytes.len() < 1 + header.length {
+            return Err(ArrowError::InvalidArgumentError(format!(
+                "Short string data length {} exceeds available bytes",
+                header.length
+            )));
+        }
+
+        Ok(&value_bytes[1..1 + header.length])
```

Comment:
```suggestion
        let Some(header) = value_bytes.get(0) else {
            return Err(ArrowError::InvalidArgumentError(
                "Empty value bytes".to_string(),
            ));
        };
        let header = Self::parse_short_string_header(header)?;
        slice_from_slice_at_offset(1, 0..header.length)
```
(simpler, and avoids integer overflow panic risk on 32-bit arch)